src/rest_vol_attr.c

```diff
@@ -479,7 +480,130 @@ RV_attr_open(void *obj, const H5VL_loc_params_t *loc_params, const char *attr_na
+                case H5I_FILE:
+                case H5I_GROUP:
+                    parent_obj_type_header = "groups";
+                    break;
+                case H5I_DATATYPE:
+                    parent_obj_type_header = "datatypes";
+                    break;
+                case H5I_DATASET:
+                    parent_obj_type_header = "datasets";
+                    break;
+                case H5I_ATTR:
+                case H5I_UNINIT:
+                case H5I_BADID:
+                case H5I_DATASPACE:
+                case H5I_VFL:
+                case H5I_VOL:
+                case H5I_GENPROP_CLS:
+                case H5I_GENPROP_LST:
+                case H5I_ERROR_CLASS:
+                case H5I_ERROR_MSG:
+                case H5I_ERROR_STACK:
+                case H5I_NTYPES:
+                default:
+                    FUNC_GOTO_ERROR(H5E_ATTR, H5E_BADVALUE, NULL,
+                                    "parent object not a group, datatype or dataset");
+            }
+
+            if ((url_len = snprintf(request_url, URL_MAX_LENGTH, "%s/%s/%s?%s&include_attrs=1", base_URL,
+                                    parent_obj_type_header, attribute->u.attribute.parent_obj_URI,
+                                    request_idx_type)) < 0)
+                FUNC_GOTO_ERROR(H5E_ATTR, H5E_SYSERRSTR, NULL, "snprintf error");
+
+            if (url_len >= URL_MAX_LENGTH)
+                FUNC_GOTO_ERROR(H5E_ATTR, H5E_SYSERRSTR, NULL,
+                                "attribute open URL exceeded maximum URL size");
+
+            if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_HTTPHEADER, curl_headers))
+                FUNC_GOTO_ERROR(H5E_ATTR, H5E_CANTSET, NULL, "can't set cURL HTTP headers: %s", curl_err_buf);
+            if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_HTTPGET, 1))
+                FUNC_GOTO_ERROR(H5E_ATTR, H5E_CANTSET, NULL, "can't set up cURL to make HTTP GET request: %s",
+                                curl_err_buf);
+            if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_URL, request_url))
+                FUNC_GOTO_ERROR(H5E_ATTR, H5E_CANTSET, NULL, "can't set cURL request URL: %s", curl_err_buf);
+
+            CURL_PERFORM(curl, H5E_ATTR, H5E_CANTGET, NULL);
+
+            /* Avoid compiler warnings about const pointer reassignment */
+            if (0 > memcpy(&idx_args, &loc_params->loc_data.loc_by_idx, sizeof(H5VL_loc_by_idx_t)))
+                FUNC_GOTO_ERROR(H5E_LINK, H5E_SYSERRSTR, NULL, "failed to copy loc by idx info");
+
+            if (0 > RV_parse_response(response_buffer.buffer, (void *)&idx_args, &found_attr_name,
+                                      RV_copy_attribute_name_by_index))
+                FUNC_GOTO_ERROR(H5E_ATTR, H5E_PARSEERROR, NULL, "failed to retrieve attribute names");
+
+            if (host_header)
+                RV_free(host_header);
+            if (url_encoded_attr_name)
+                curl_free(url_encoded_attr_name);
```

Comment:
These two variables should be set to NULL after freeing so we don't accidentally double-free them